app/routes.py

```diff
@@ -1,2 +1,208 @@
+        "text" : f"Task {task.task_id} with title {task.title} has been marked as complete"
+    }
+    authorization = os.environ.get('SLACK_TOKEN')
+    headers = {"Authorization": f"Bearer {authorization}"}
+
+    response = requests.post(path, data=query_params, headers=headers)
+    print(response.text)
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+    
+    task.completed_at = None
+    db.session.commit()
+
+    response = {"task":task.task_response()}
+    return jsonify(response), 200
+
+
+#### Goals ####
+
+@goals_bp.route("", methods=["POST","GET"])
+def handle_goals():
+    if request.method == "POST":
+        request_body = request.get_json()
+        if 'title' in request_body: 
+            new_goal = Goal(title=request_body["title"])
+
+            db.session.add(new_goal)
+            db.session.commit()
+
+            response = {"goal": new_goal.goal_response()}
+            return jsonify(response), 201
+        else:
+            return make_response ({"details": "Invalid data"},400)
+
+    elif request.method == "GET":
+        goals = Goal.query.all()
+    
+        goals_response = []
+        
+        for goal in goals:
+            goals_response.append(goal.goal_response())
+        return jsonify(goals_response),200
+
+@goals_bp.route("/<goal_id>", methods=["GET","PUT","DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+   
+    if goal is None:
+        return make_response("", 404)
+
+    elif request.method == "GET":
+        response = {"goal":goal.goal_response()}
+        return jsonify(response), 200
+
+    elif request.method == "PUT":      
+        form_data = request.get_json()
+        goal.title = form_data["title"]
+
+        db.session.commit()
+        
+        response = {"goal":goal.goal_response()}
+        return jsonify(response), 200
+        
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        delete_text = f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"
+        response = {"details":delete_text}
+        return jsonify(response), 200
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET", "POST"])
+def handle_goal_tasks(goal_id):
+        goal = Goal.query.get(goal_id)
+
+        if goal is None:
+            return make_response("", 404)
+
+        elif request.method =="POST":
+            request_body = request.get_json()
+
+            for task_id in request_body["task_ids"]:
+                tasks = []
+                tasks.append(Task.query.get(task_id))
+
+                for task in tasks:
+                    task.goal_id = goal_id
+                    db.session.commit()
+
+            response = {}
+            response["id"] = int(goal_id)
+            response["task_ids"] = request_body["task_ids"]
```

Comment:
This change would have made the above-mentioned bug more obvious. This would also allow the response to include ALL tasks associated with the goal, not just the new ones.
This isn't super intuitive so I'd definitely be down to talk through this more in our next 1:1

```suggestion
            full_task_ids = []
            for task in goal.tasks:
                full_task_ids.append(task.task_id)
        
             response["task_ids"] = full_task_ids
```